src/solvers/dg/2d/dg.jl

```diff
@@ -1530,43 +1563,152 @@ end
 
 
 """
-    calcflux_fv!(fstar1, fstar2, u_leftright, u, element_id, dg::Dg2D)
+    calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u_leftright, u, element_id, dg::Dg2D)
 
-Calculate the finite volume fluxes inside the elements.
+Calculate the finite volume fluxes inside the elements (without non-conservative terms).
 
 # Arguments
-- `fstar1::AbstractArray{T} where T<:Real`:
-- `fstar2::AbstractArray{T} where T<:Real`
+- `fstar1_L::AbstractArray{T} where T<:Real`:
+- `fstar1_R::AbstractArray{T} where T<:Real`:
+- `fstar2_L::AbstractArray{T} where T<:Real`:
+- `fstar2_R::AbstractArray{T} where T<:Real`:
 - `dg::Dg2D`
 - `u::AbstractArray{T} where T<:Real`
 - `element_id::Integer`
 """
-@inline function calcflux_fv!(fstar1, fstar2, u, element_id, dg::Dg2D)
+@inline function calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u, element_id, dg::Dg2D, nonconservative_terms::Val{false})
   @unpack surface_flux_function = dg
 
-  fstar1[:, 1,            :] .= zero(eltype(fstar1))
-  fstar1[:, nnodes(dg)+1, :] .= zero(eltype(fstar1))
+  fstar1_L[:, 1,            :] .= zero(eltype(fstar1_L))
+  fstar1_L[:, nnodes(dg)+1, :] .= zero(eltype(fstar1_L))
+  fstar1_R[:, 1,            :] .= zero(eltype(fstar1_R))
+  fstar1_R[:, nnodes(dg)+1, :] .= zero(eltype(fstar1_R))
 
   for j in 1:nnodes(dg)
     for i in 2:nnodes(dg)
       u_ll = get_node_vars(u, dg, i-1, j, element_id)
       u_rr = get_node_vars(u, dg, i,   j, element_id)
       flux = surface_flux_function(u_ll, u_rr, 1, equations(dg)) # orientation 1: x direction
-      set_node_vars!(fstar1, flux, dg, i, j)
+      set_node_vars!(fstar1_L, flux, dg, i, j)
+      set_node_vars!(fstar1_R, flux, dg, i, j)
     end
   end
 
-  fstar2[:, :, 1           ] .= zero(eltype(fstar2))
-  fstar2[:, :, nnodes(dg)+1] .= zero(eltype(fstar2))
+  fstar2_L[:, :, 1           ] .= zero(eltype(fstar2_L))
+  fstar2_L[:, :, nnodes(dg)+1] .= zero(eltype(fstar2_L))
+  fstar2_R[:, :, 1           ] .= zero(eltype(fstar2_R))
+  fstar2_R[:, :, nnodes(dg)+1] .= zero(eltype(fstar2_R))
 
   for j in 2:nnodes(dg)
     for i in 1:nnodes(dg)
       u_ll = get_node_vars(u, dg, i, j-1, element_id)
       u_rr = get_node_vars(u, dg, i, j,   element_id)
       flux = surface_flux_function(u_ll, u_rr, 2, equations(dg)) # orientation 2: y direction
-      set_node_vars!(fstar2, flux, dg, i, j)
+      set_node_vars!(fstar2_L, flux, dg, i, j)
+      set_node_vars!(fstar2_R, flux, dg, i, j)
     end
   end
+  
+end
+
+"""
+    calcflux_fv!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u_leftright, u, element_id, dg::Dg2D)
+
+Calculate the finite volume fluxes inside the elements (non-conservative terms are active).
```

Comment:
```suggestion
Calculate the finite volume fluxes inside the elements (**with non-conservative terms**).
```
To be symmetric to the function above